lib/flagutil/map.go

```diff
@@ -0,0 +1,37 @@
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+)
+
+type MapString map[string]string
+
+func (m *MapString) String() string {
+	return fmt.Sprintf("%v", *m)
+}
+
+func (m *MapString) Set(value string) error {
+	if *m == nil {
+		*m = make(map[string]string)
+	}
+	for _, pair := range parseArrayValues(value) {
+		key, value := parseMapValue(pair)
+		(*m)[key] = value
+	}
+	return nil
+}
+
+func parseMapValue(s string) (string, string) {
+	kv := strings.SplitN(s, ":", 2)
+	return kv[0], kv[1]
+}
+
+// NewMapString returns a new MapString with the given name and description.
+func NewMapString(name, description string) *MapString {
+	description += "\nSupports multiple flags with the following syntax: -name=key:value"
```

Comment:
as far you know the name why don't use it in description
Supports multiple flags with the following syntax: -name=key:value -> Supports multiple flags with the following syntax: -%s=key:value (where %s -> name variable)